project-generation/helpers.go

```diff
@@ -193,21 +272,45 @@ func promptForConfirmation(ctx context.Context, prompt string, maxInputAttempts
 	return false
 }
 
-func promptForInput(ctx context.Context, prompt string) (string, error) {
+// PromptForInput will write a line to output then wait for input which is returned from the function
+func PromptForInput(ctx context.Context, prompt string) (string, error) {
 	var input string
 	scanner := bufio.NewScanner(os.Stdin)
 	fmt.Println(prompt)
 	scanner.Scan()
 	input = scanner.Text()
 	if scanner.Err() != nil {
-		log.Event(ctx, "project creation failed", log.Error(scanner.Err()))
+		log.Event(ctx, "Failed to read user input", log.Error(scanner.Err()))
 		return "", scanner.Err()
 	}
 	return input, nil
 }
 
-// initGoModules will initialise the go modules for a project at a given directory
-func initGoModules(ctx context.Context, pathToRepo, name string) {
+func OptionPromptInput(ctx context.Context, prompt string, options ...string) (string, error) {
+	var input string
+	var optionSelected int
+	scanner := bufio.NewScanner(os.Stdin)
+	fmt.Println(prompt + "\n")
+	for i, option := range options {
+		fmt.Printf("[%d] %v \n", i, option)
+	}
+	fmt.Println("\nPlease enter the number corresponding to your choice:")
+	scanner.Scan()
+	input = scanner.Text()
+	if scanner.Err() != nil {
+		log.Event(ctx, "failed to read user input", log.Error(scanner.Err()))
+		return "", scanner.Err()
+	}
+	optionSelected, err := strconv.Atoi(input)
+	if scanner.Err() != nil {
+		log.Event(ctx, "failed to convert user input to valid option", log.Error(err))
+		return "", scanner.Err()
+	}
```

Comment:
Is there a case here where a user could enter text other than a number, or a number larger than the possible options?